wandb/sdk/data_types/base_types/media.py

```diff
@@ -236,6 +237,10 @@ def to_json(self, run: Union["LocalRun", "LocalArtifact"]) -> dict:
                         self._path
                     ):
                         artifact.add_reference(self._path, name=name)
+                    elif isinstance(self, Image) and Image.path_is_reference(
```

Comment:
This looks very similar to the block above - there might be an opportunity to refactor this.